File: Recursion/Example/Tylor_Series.java
package Recursion.Example;


class Tylor_Horner_Rule {
    // using recursion //
    static double result;
    protected double Tylor_Hons_Rule(int x, int n) {
        if (x == 0)
            return result;
        result = 1 + (double)(x/n)*result;
        return Tylor_Hons_Rule(x, n-1);
    }



}


public class Tylor_Series {
    // using for loop
    private static double tylor(int x, int n) {
        double res = 1, num = 1, den = 1;
        for (int i = 1; i<= n; i++) {
            num *= x;
            den *= i;
            res += num/den;
        }
        return res;
    }
    static int p = 1, f = 1;

    public static double tylor_series(int x, int n) { 
        double result;
        if (n == 0) {
            return 1;
        } else {
            result = tylor_series(x, n - 1);
            p *= x;
            f *= n;
            return result + (double)p/f;
        }
    }



    public static void main(String[] args) {
        int x = 1;
        int y = 10;
        System.out.println(tylor_series(x, y));

        // Tylor Using Horner's Rule //
//        Tylor_Horner_Rule tr = new Tylor_Horner_Rule();
//        double res = tr.Tylor_Hons_Rule(1,10);  // error
        System.out.println(tylor(x,y));

    }
}

File: Recursion/Examples/tylorSeries.cpp
#include <iostream>
using namespace std;

// way 1 : 

double tylor_series(int x, int n) {
    double ans = 0;
    static double p = 1, f = 1;

    if (n == 0) {
        return 1;
    } else {
        ans = tylor_series(x, n-1);
        p *= x;
        f *= n;
        return ans + p/f;
    }
}

// Way 2 : (Horner's Rule) 
// Iterative //

double tylor_iter(int x, int n) {
    double ans = 1;
    for (; n > 0; --n) {
        ans = 1 + x*ans/n;
    }
    return ans;
}

// using recursion //

double tylor_recursion(int x, int n) {
    static double ans = 1;
    if (n == 0) {
        return ans;
    }

    ans = 1 + x*ans/n;
    return tylor_recursion(x, n-1);
}


double tylor(int x, int n) {
    double ans = 1;
    double num = 1, den = 1;

    for (int i = 1; i <= n; ++i) {
        num *= x;
        den *= i;
        ans += num/den;
    }

    return ans;
}
int main() {

    double x, n;
    double ans;
    cout << "Enter The Value Of X : ";
    cin >> x;----------------------------------------------------------------
    cout << "Enter The Value Of N : ";
    cin >> n;

    ans = tylor(x, n);
    cout << "Using Tylor Series : (Exponent Value Of x and n means x^n is :) "  << ans << endl;
    return 0;
}